app/services/store_sp_metadata_in_session.rb

```diff
@@ -5,6 +5,8 @@ def initialize(session:, request_id:)
   end
 
   def call
+    return unless sp_request
```

Comment:
Yeah, see comment above. I'm glad that somebody else thought silently eating the error was a bad idea too.

I think we can catch this error earlier on by validating the request ID when we validate the confirmation token